src/app/common/services/data.service.ts

```diff
@@ -0,0 +1,138 @@
+import { Student } from '../entities/student';
+import { STUDENTS } from '../constants/constants-student';
+
+import { Subject } from '../entities/subject';
+import { SUBJECTS } from '../constants/constants-subject';
+
+import { Teacher } from '../entities/teacher';
+import { TEACHER } from '../constants/constants-teacher';
+
+import { Mark } from '../entities/mark';
+import { StudentMark } from '../entities/student-mark';
+
+export class DataService {
+
+  private dataStudents: Student[] = STUDENTS;
+  private dataSubjects: Subject[] = SUBJECTS;
+  private dataTeachers: Teacher[] = TEACHER;
+
+  getDataStudents(): Student[] {
+    return this.dataStudents;
+  }
+
+  getDataTeachers(): Teacher[] {
+    return this.dataTeachers;
+  }
+
+  getDataStudent(studentId: number) {
+    return this.dataStudents.find(student => student.id === studentId);
+  }
+
+  addDataNewStudent(
+    name: string,
+    lastName: string,
+    age: number,
+    address: string
+  ): void {
+    if (name == null || lastName == null || name.trim() == "" || age == null)
```

Comment:
Why not triple equal?
Can we check fields this way: (!name || !lastName || !name.trim() || !age) ?